packages/core/src/components/DataTable/types.ts

```diff
@@ -0,0 +1,224 @@
+  /** If enabled, clicking the row triggers the same function as click the selection checkbox. */
+  selectOnRowClick?: boolean;
+  /** If enabled, renders a border between each column. */
+  showColumnDividers?: boolean;
+  /** If enabled, renders a border between each row. */
+  showRowDividers?: boolean;
+  /** Overrides sort and hands control to the parent component. */
+  sortOverride?: boolean;
+  /** sortBy value if override is enabled. */
+  sortByOverride?: string;
+  /** sortDirection value if override is enabled. */
+  sortDirectionOverride?: SortDirectionType;
+  /** sortCallback in override is enabled. */
+  sortCallback?: (sortBy: string, sortDirection: SortDirectionType) => void;
+  /** Label for the table header. */
+  tableHeaderLabel?: string;
+  /** Height of the table header. */
+  tableHeaderHeight?: HeightOptions;
+  /** Width of the entire table. */
+  width?: number;
+  /** If enabled, every other row will appear in grey. */
+  zebra?: boolean;
+}
+
+export interface GenericRow {
+  data: {
+    [key: string]: any;
+  };
+}
+
+/** The row used by React Virtualized. */
+export interface TableRow {
+  cellData?: void;
+  columnData?: void;
+  columnIndex: number;
+  dataKey: string;
+  isScrolling?: boolean;
+  parent?: React.ReactNode;
+  rowData: ExpandedRow;
+  rowIndex: number;
+}
+
+export type Status = string;
+
+type ParentMetadata = {
+  colSpanKey?: string;
+  children?: ChildRow[];
+  status?: string;
+};
+
+type ChildMetadata = {
+  colSpanKey?: string;
+  status?: Status;
+};
+
+type IndexedParentMetadata = ParentMetadata & {
+  originalIndex: number;
+  isChild: boolean;
+  children: IndexedChildRow[];
+};
+
+type IndexedChildMetadata = ChildMetadata & {
+  originalIndex: number;
+  children?: undefined;
+  isChild: boolean;
+};
+
+type ExpandedParentMetadata = IndexedParentMetadata & {
+  preExpandedIndex: number;
+  parentIndex?: undefined;
+  parentOriginalIndex?: undefined;
+};
+
+type ExpandedChildMetadata = IndexedChildMetadata & {
+  preExpandedIndex?: undefined;
+  parentIndex: number;
+  parentOriginalIndex: number;
+};
+
+interface Row {
+  data: {
+    [key: string]: any;
+  };
+}
+
+export interface ChildRow extends Row {
+  metadata?: ChildMetadata;
+}
+
+export interface ParentRow extends Row {
+  metadata?: ParentMetadata;
+}
+
+export interface IndexedParentRow extends Row {
+  metadata: IndexedParentMetadata;
+}
+export interface IndexedChildRow extends Row {
+  metadata: IndexedChildMetadata;
+}
+
+export interface ExpandedParentRow extends Row {
+  metadata: ExpandedParentMetadata;
+}
+export interface ExpandedChildRow extends Row {
+  metadata: ExpandedChildMetadata;
+}
+
+export type IndexedRow = IndexedChildRow | IndexedParentRow;
+export type ExpandedRow = ExpandedParentRow | ExpandedChildRow;
+
+export type RowStyles = {
+  [key: string]: string;
+};
+
+export type ColumnMetadata = {
+  [key: string]: {
+    [key: string]: number;
+  };
+};
+
+export type ColumnToLabel = {
```

Comment:
If this is the primary hook for rendering `ReactNode`s in column headers, could we update the type to `ReactNode`?

Reply:
The type is now `any`